src/lib.rs.in

```diff
@@ -136,9 +137,14 @@ pub struct AuthHandler { pub app: AppConfig }
 impl Handler for AuthHandler {
     fn handle(&self, req: &mut Request) -> IronResult<Response> {
         info!("{} {}", req.method, req.url);
-        let params = try!(req.get_ref::<UrlEncodedBody>().map_err(
-                          |e| IronError::new(e, (status::BadRequest,
-                                                 "no query string in POST data"))));
+        let params = try!(
+            match req.method {
+                Method::Get => req.get_ref::<UrlEncodedQuery>(),
+                _ => req.get_ref::<UrlEncodedBody>()
```

Comment:
I agree, but couldn't think a way out of this one when I wrote it. Now it seems obvious. :)